### ISSUE TITLE
Lacking API to set material/texture for object

### ISSUE THREAD

**Issue by rongguodong** [Samsung/GearVRf#172]
Currently, if the user want to set material/texture for an object, he/she should do like following:
object.getRenderData().setMaterial(material); or
object.getRenderData().getMaterial().setMainTexture(texture);

It should be good to add new APIs in GVRSceneObject to set material/texture directly. The advantage of this approach are:
1. The user do not need to care about inner data structure (e.g. RenderData);
2. The new API can recursively apply the material/texture to all children objects and their decedents if necessary. This is difficult or impossible to be done by the user.

**Comment by phi-lira** [comment]
I agree with adding setMaterial/SetTexture interface to SceneObject. 

About setting it recursively I think that should be discussed with more care. If done, it should definitely not be default behaviour as this can be error prone.

**Comment by rongguodong** [comment]
Sure, the recursive setting should not be a default behavior. We should have a boolean variable to control it, and it should be set to "false" by default. The user can call an API to set the boolean variable.

**Comment by phi-lira** [comment]
It's ok for me.

**Comment by tparisi** [comment]
I think it shouldn't be done recursively. That's a pretty unusual pattern for a scene graph API. It might be better to have scene traversal iterators that can be called recursively so someone could build that as a utility if they really need it.

**Comment by thomasflynn** [comment]
I agree with @tparisi .  Since GVRSceneObject is treated as both a drawable and a group object, you want to approach this with caution.  For the derived SceneObjects (like GVRSphereSceneObject and friends), there should definitely be some convenience methods that 'just does the right thing' for setting a texture/material on that object.  Those derived objects know about their internals and will know how to apply them appropriately.

Doing it recursively, even as a flag, sounds like we're setting ourselves up for a lot of undefined behavior to happen.
